index.js

```diff
@@ -58,7 +58,11 @@ function redirect(options, ctx) {
   }
 
   // build redirect url
-  const httpsHost = options.hostname || url.parse('http://' + ctx.request.header.host).hostname;
+  const httpsHost = (
+    options.hostname &&
+    typeof options.hostname === 'function' ? options.hostname(ctx) : options.hostname
```

Comment:
I would maybe pass only the host to the function and expect only the new host or null/undefined back. Just to make it a bit easier to understand and harder to mess things up. If someone needs access to whole `ctx` then I think implementing another middleware would be a better option - that way the control flow (order of middlewares) would be more explicit.

Reply:
Using `request.header.host = 'foo.com:8080'` as an example, should we provide the `host` (e.g. `foo.com:8080`) or `hostname` (e.g. `foo.com`)? I suppose that `host` is the more complete one, and I will go with that.

Reply:
Derr...now I see that this lib accomodates `port` on its own. I'll go with `hostname` :)

Reply:
:D no worries. Just ping me once you have a version you're happy with ;)